desc/objectives/_stability.py

```diff
@@ -367,3 +369,275 @@ def compute(self, params, constants=None):
+        )
+        fieldline_nodes = np.array([rho, alpha, zeta]).T
+
+        self._dim_f = 1
+        self._data_keys = ["ideal_ball_gamma2"]
+
+        self._args = get_params(
+            self._iota_keys + self._len_keys + self._data_keys,
+            obj="desc.equilibrium.equilibrium.Equilibrium",
+            has_axis=False,
+        )
+
+        self._constants = {
+            "iota_transforms": iota_transforms,
+            "iota_profiles": iota_profiles,
+            "len_transforms": len_transforms,
+            "len_profiles": len_profiles,
+            "fieldline_nodes": fieldline_nodes,
+            "N_alpha": int(8),
+            "N_zeta": self.nzeta,
+            "quad_weights": 1.0,
+        }
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """
+        Compute the ballooning stability growth rate.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of equilibrium degrees of freedom, eg Equilibrium.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        gamma : ndarray
+            ideal ballooning growth rate.
+
+        """
+        from desc.equilibrium.coords import map_coordinates
+
+        eq = self.things[0]
+
+        if constants is None:
+            constants = self.constants
+        # we first compute iota on a uniform grid to get correct averaging etc.
+        iota_data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._iota_keys,
+            params=params,
+            transforms=constants["iota_transforms"],
+            profiles=constants["iota_profiles"],
+        )
+
+        len_data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._len_keys,
+            params=params,
+            transforms=constants["len_transforms"],
+            profiles=constants["len_profiles"],
+        )
+
+        # Now we compute theta_DESC for given theta_PEST
+        iota = iota_data["iota"][0]
+        rho, alpha, zeta = constants["fieldline_nodes"].T
+        theta_PEST = alpha + iota * zeta
+        nodes = jnp.array([rho, theta_PEST, zeta]).T
+
+        # we prime the data dict with the correct iota values so we don't recompute them
+        # using the wrong grid
+        data = {
+            "iota": iota_data["iota"][0] * jnp.ones_like(zeta),
+            "iota_r": iota_data["iota_r"][0] * jnp.ones_like(zeta),
+            "shear": iota_data["shear"][0] * jnp.ones_like(zeta),
+            "a": len_data["a"],
+        }
+
+        N_alpha = constants["N_alpha"]
+        N_zeta = constants["N_zeta"]
+
+        # Rootfinding theta for a given theta_PEST
+        desc_coords = map_coordinates(eq, nodes, inbasis=("rho", "theta_PEST", "zeta"))
+
+        # DIFFERENT WAYS THAT DON'T WORK!
```

Comment:
Probably we want to remove these before merging in, you can make a comment on the PR with these ways that don't work but they aren't needed in the code (in my opinion at least but if others are fine with it then that is fine)

Reply:
Removed.